libraries/AP_HAL_Empty/UARTDriver.cpp

```diff
@@ -17,13 +17,7 @@ uint32_t Empty::UARTDriver::txspace() { return 1; }
 int16_t Empty::UARTDriver::read() { return -1; }
 
 /* Empty implementations of Print virtual methods */
-size_t Empty::UARTDriver::write(uint8_t c) { return 0; }
-
 size_t Empty::UARTDriver::write(const uint8_t *buffer, size_t size)
 {
-    size_t n = 0;
-    while (size--) {
-        n += write(*buffer++);
-    }
-    return n;
+    return -1;
```

Comment:
this doesn't    seem right, particularly because return is unsigned. We should probably return 0 here.

Reply:
or `size` if you want to change it to "yeah, we wrote everything"

Reply:
Changed it to zero - I can't see it really matters one way of the other and this would seem to preserve existing behaviour....

Thanks.